server/src/main/java/org/elasticsearch/common/time/CompoundDateTimeFormatter.java

```diff
@@ -70,4 +76,68 @@ public CompoundDateTimeFormatter withZone(ZoneId zoneId) {
     public String format(TemporalAccessor accessor) {
         return printer.format(accessor);
     }
+
+    public static CompoundDateTimeFormatterBuilder builder() {
+        return new CompoundDateTimeFormatterBuilder();
+    }
+
+    static final class CompoundDateTimeFormatterBuilder {
```

Comment:
Instead of creating another class, could the functionality here just be a couple private static methods on DateFormatters? If I understand these timezone formats correctly, the only reason we have any of this complexity is because of `+HHmm`, which is a non standard format. The other two would be handled by `appendZoneOrOffsetId()`? I think it would be good to keep this local to DateFormatters, so that adding more cases like this does not leak outside of that class.